File: Process Names/check_all_process_names.py
#
#   Checks all loaded process names, Python
#   Module written by Brandon Arvanaghi
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

import win32pdh

EvidenceOfSandbox = []
sandboxProcesses = "vmsrvc", "tcpview", "wireshark", "visual basic", "fiddler", "vmware", "vbox", "process explorer", "autoit", "vboxtray", "vmtools", "vmrawdsk", "vmusbmouse", "vmvss", "vmscsi", "vmxnet", "vmx_svga", "vmmemctl", "df5serv", "vboxservice", "vmhgfs"

_, runningProcesses = win32pdh.EnumObjectItems(None,None,'process', win32pdh.PERF_DETAIL_WIZARD)

for process in runningProcesses:
    for sandboxProcess in sandboxProcesses:
        if sandboxProcess in str(process):
            EvidenceOfSandbox.append(process)
            break

if not EvidenceOfSandbox:
    print("Proceed!")
else:
    print(EvidenceOfSandbox)

File: Process Names/check_all_process_names.pl
#
#   Checks all loaded process names, Perl
#   Module written by Brandon Arvanaghi 
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

use Win32::OLE;

my @EvidenceOfSandbox;
my @sandboxProcesses = (q{vmsrvc}, q{tcpview}, q{wireshark}, q{visual basic}, q{fiddler}, q{vmware}, q{vbox}, q{process explorer}, q{autoit}, q{vboxtray}, q{vmtools}, q{vmrawdsk}, q{vmusbmouse}, q{vmvss}, q{vmscsi}, q{vmxnet}, q{vmx_svga}, q{vmmemctl}, q{df5serv}, q{vboxservice}, q{vmhgfs});

my $runningProcesses = Win32::OLE->GetObject("winmgmts:\\\\localhost\\root\\CIMV2")->ExecQuery("SELECT * FROM Win32_Process") or die;

foreach my $process (in $runningProcesses) {
	foreach $sandboxProcess (@sandboxProcesses) {
		if (index(lc($process->{Name}), lc($sandboxProcess)) != -1) {
			if (!(grep $_ eq $process->{Name}, @EvidenceOfSandbox)) {
				push(@EvidenceOfSandbox, $process->{Name});
			}
		} 
	}
}

if (!@EvidenceOfSandbox) {
	print "No sandbox-indicative process name was found running on the system. Proceed!\n";
} else {
	print "The following running processes suggest we are running in a sandbox. Do not proceed.\n", join("\n", @EvidenceOfSandbox), "\n";
}